src/graphene.cpp

```diff
@@ -0,0 +1,1504 @@
+string CGrapheneBlockData::InBoundPercentToString()
+{
+    LOCK(cs_graphenestats);
+
+    expireStats(mapGrapheneBlocksInBound);
+
+    double nCompressionRate = 0;
+    uint64_t nGrapheneSizeTotal = 0;
+    uint64_t nOriginalSizeTotal = 0;
+    for (map<int64_t, pair<uint64_t, uint64_t> >::iterator mi = mapGrapheneBlocksInBound.begin();
+         mi != mapGrapheneBlocksInBound.end(); ++mi)
+    {
+        nGrapheneSizeTotal += (*mi).second.first;
+        nOriginalSizeTotal += (*mi).second.second;
+    }
+    // We count up the outbound CMemPoolInfo sizes. Outbound CMemPoolInfo sizes go with Inbound graphene blocks.
+    uint64_t nOutBoundMemPoolInfoSize = 0;
+    for (map<int64_t, uint64_t>::iterator mi = mapMemPoolInfoOutBound.begin(); mi != mapMemPoolInfoOutBound.end(); ++mi)
+    {
+        nOutBoundMemPoolInfoSize += (*mi).second;
+    }
+
+    if (nOriginalSizeTotal > 0)
+        nCompressionRate = 100 - (100 * (double)(nGrapheneSizeTotal + nOutBoundMemPoolInfoSize) / nOriginalSizeTotal);
+
+    ostringstream ss;
+    ss << fixed << setprecision(1);
+    ss << "Compression for " << mapGrapheneBlocksInBound.size()
+       << " Inbound  graphene blocks (last 24hrs): " << nCompressionRate << "%";
+
+    return ss.str();
+}
+
+// Calculate the graphene percentage compression over the last 24 hours
+string CGrapheneBlockData::OutBoundPercentToString()
+{
+    LOCK(cs_graphenestats);
+
+    expireStats(mapGrapheneBlocksOutBound);
+
+    double nCompressionRate = 0;
+    uint64_t nGrapheneSizeTotal = 0;
+    uint64_t nOriginalSizeTotal = 0;
+    for (map<int64_t, pair<uint64_t, uint64_t> >::iterator mi = mapGrapheneBlocksOutBound.begin();
+         mi != mapGrapheneBlocksOutBound.end(); ++mi)
+    {
+        nGrapheneSizeTotal += (*mi).second.first;
+        nOriginalSizeTotal += (*mi).second.second;
+    }
+    // We count up the inbound CMemPoolInfo sizes. Inbound CMemPoolInfo sizes go with Outbound graphene blocks.
+    uint64_t nInBoundMemPoolInfoSize = 0;
+    for (map<int64_t, uint64_t>::iterator mi = mapMemPoolInfoInBound.begin(); mi != mapMemPoolInfoInBound.end(); ++mi)
+        nInBoundMemPoolInfoSize += (*mi).second;
+
+    if (nOriginalSizeTotal > 0)
+        nCompressionRate = 100 - (100 * (double)(nGrapheneSizeTotal + nInBoundMemPoolInfoSize) / nOriginalSizeTotal);
+
+    ostringstream ss;
+    ss << fixed << setprecision(1);
+    ss << "Compression for " << mapGrapheneBlocksOutBound.size()
+       << " Outbound graphene blocks (last 24hrs): " << nCompressionRate << "%";
+    return ss.str();
+}
+
+// Calculate the average inbound graphene CMemPoolInfo size
+string CGrapheneBlockData::InBoundMemPoolInfoToString()
+{
+    LOCK(cs_graphenestats);
+    double avgMemPoolInfoSize = average(mapMemPoolInfoInBound);
+    ostringstream ss;
+    ss << "Inbound CMemPoolInfo size (last 24hrs) AVG: " << formatInfoUnit(avgMemPoolInfoSize);
+    return ss.str();
+}
+
+// Calculate the average inbound graphene CMemPoolInfo size
```

Comment:
s/inbound/outbound in the comment. 
(guess the same error is present in `src/thinblock.cpp`)

Reply:
Done.